lib/components/page/default_grid_section.dart

```diff
@@ -0,0 +1,326 @@
+                child: featuredTag!,
+              ),
+          ],
+        ),
+      ),
+    );
+  }
+
+  Widget? get featuredTag {
+    if (isLive) {
+      return const FeatureTag(
+        label: 'Live now',
+        color: Color.fromRGBO(230, 60, 98, 1),
+      );
+    } else if (isComingSoon) {
+      return const FeatureTag(
+        label: 'Coming soon',
+        color: Color.fromRGBO(29, 40, 56, 1),
+      );
+    } else if (isNewItem) {
+      return const FeatureTag(
+        label: 'New',
+        color: Color.fromRGBO(230, 60, 98, 1),
+      );
+    }
+    return null;
+  }
+}
+
+class _GridShowItem extends StatelessWidget {
+  final Fragment$Section$$DefaultGridSection$items$items sectionItem;
+  final Fragment$Section$$DefaultGridSection$items$items$item$$Show show;
+
+  // TODO: Remove this
+  bool hasNewEpisodes = false;
+
+  _GridShowItem(this.sectionItem)
+      : show = sectionItem.item
+            as Fragment$Section$$DefaultGridSection$items$items$item$$Show;
+
+  @override
+  Widget build(BuildContext context) {
+    return Column(
+      crossAxisAlignment: CrossAxisAlignment.start,
+      children: [
+        sectionItemImage,
+        Container(
+          margin: const EdgeInsets.only(bottom: 2),
+          child: Text(
+            sectionItem.title,
+            maxLines: 2,
+            overflow: TextOverflow.ellipsis,
+            style: const TextStyle(
+              fontSize: 14,
+              fontWeight: FontWeight.w500,
+            ),
+          ),
+        ),
+        Text(
+          '${show.seasonCount} Seasons - ${show.episodeCount} Episodes',
+          style: const TextStyle(
+            fontSize: 12,
+            fontWeight: FontWeight.w500,
+            color: Color.fromRGBO(112, 124, 142, 1),
+          ),
+        )
+      ],
+    );
+  }
+
+  Widget get sectionItemImage {
+    return Container(
+      margin: const EdgeInsets.only(bottom: 4),
+      width: double.infinity,
+      child: AspectRatio(
+        aspectRatio: 16 / 9,
+        child: Stack(
+          clipBehavior: Clip.none,
+          children: [
+            BorderedImageContainer(
+              image: sectionItem.image != null
+                  ? NetworkImage(sectionItem.image!)
+                  : null,
+            ),
+            if (hasNewEpisodes)
+              const Positioned(
+                top: -4,
+                right: -4,
+                child: FeatureTag(
+                  label: 'New Episodes',
+                  color: Color.fromRGBO(230, 60, 98, 1),
+                ),
+              ),
+          ],
+        ),
+      ),
+    );
+  }
+}
+
+class _GridRow extends StatelessWidget {
+  final int colSize;
+  final List<Widget> items;
+
+  const _GridRow({required this.items, required this.colSize});
+
+  List<Widget> get spacedItems {
+    if (items.length < colSize) {
+      items.addAll(
+        List<Container>.generate(
+            colSize - items.length, (index) => Container()),
```

Comment:
Grid view requires that all the children have the same width and height which needs to be specified by aspect ratio. In our case, since the text can be 1 or 2 lines, it's a bit hacky to calculate I think.